packages/stack/src/views/Stack/CardContainer.tsx

```diff
@@ -120,11 +129,22 @@ function CardContainer({
     onCloseRoute({ route: scene.route });
   };
 
+  const handleGestureBegin = () => {
+    onPageChangeStart?.();
+    onGestureStart?.({ route: scene.route });
+  };
+
+  const handleGestureCanceled = () => {
+    onPageChangeCancel?.();
+    onGestureCancel?.({ route: scene.route });
+  };
+
   const handleTransitionStart = ({ closing }: { closing: boolean }) => {
     if (active && closing) {
       onPageChangeConfirm?.();
+      onGestureEnd?.({ route: scene.route });
     } else {
-      onPageChangeCancel?.();
+      handleGestureCanceled();
```

Comment:
starting a transition shouldn't assume that a gesture was cancelled

Reply:
Ok, makes sense. Reverting...